### ISSUE TITLE
Add logging to capture callback from VA Notify when a user has received a ZSF email

### ISSUE THREAD

**Issue by andrea-merrill** [department-of-veterans-affairs/va.gov-team#98603]
Now that VA Notify supports Service Callback Options, we want to capture the status of our ZSF email communications to users.  Specifically, we want to know if the email reach the intended recipient. We'll also need to add a new logging tag so that it's captured in Lighthouse as well.

## Implementation details
- [New documentation](https://github.com/department-of-veterans-affairs/vets-api/tree/master/modules/va_notify) for VA Notify
- [Logging documentation](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/platform/practices/zero-silent-failures/logging-silent-failures.md) from Platform/Lighthouse

## Acceptance Criteria
- We're logging when a ZSF email has reached the intended recipient 
- We added the logging tag so that this information is passed on to Lighthouse friends

**Comment by michaelclement** [comment]
As additional context, this should be pretty straightforward - our logging function takes an extra parameter (`email_confirmed`), so in [this instance](https://github.com/department-of-veterans-affairs/vets-api/blob/0ce65f5ad64c1edad97b2060cf82730734901505/modules/ivc_champva/app/jobs/ivc_champva/missing_form_status_job.rb#L33) we would just need to pass the result of the callback alongside the additional_context param.

**Comment by michaelclement** [comment]
To expand on the above, the actual logging should probably happen inside the callback function itself, which would require a slight refactor to the existing flow.

**Comment by michaelclement** [comment]
Experimenting in [this PR](https://github.com/department-of-veterans-affairs/vets-api/pull/20015), not quite working; still trying to rough in the details.

**Comment by michaelclement** [comment]
Still having no luck. After pairing with Steve, we decided that it would be worth asking the VA Notify folk. [Slack thread here](https://dsva.slack.com/archives/C010R6AUPHT/p1735681070281949).

**Comment by michaelclement** [comment]
Plan now is to merge into staging to test more fully there. Waiting on engineering approvals.

**Comment by michaelclement** [comment]
This was merged this morning, should be in staging shortly.

**Comment by michaelclement** [comment]
Did some more testing in staging. Looks like there's a potential issue coming from VA Notify where the statuses we rely on for the callback aren't being updated. I added a [comment](https://dsva.slack.com/archives/C010R6AUPHT/p1736456628257749?thread_ts=1735681070.281949&cid=C010R6AUPHT) to this thread in the VA Notify chat and am hoping to get some guidance.

**Comment by michaelclement** [comment]
Got some useful info from VA Notify team - apparently they had to enable something on our end for the statuses to come through. There's a new issue where the callback doesn't seem to fire when it should be automatically triggered, but going to have to look into it tomorrow.

**Comment by michaelclement** [comment]
Finally, we have email confirmation for when our ZSF emails are sent:

![Image](https://github.com/user-attachments/assets/47906b96-b6ff-4665-8fb1-47cc7f40167c)

**Comment by michaelclement** [comment]
@baltes This has been merged into staging and should go to production with tomorrow's deploy. There's a [feature toggle](https://api.va.gov/flipper/features/champva_vanotify_custom_callback) we need to enable in production for this to start working out there, but you can see from the screenshot above, it works on staging. 

Please let me know if there's any other information you'd like me to provide on this one.
